lib/core/networking/websockets/middleware/encryption_middleware.dart

```diff
@@ -0,0 +1,216 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:cryptography/cryptography.dart';
+import 'package:synchronized/extension.dart';
+
+import 'middleware_types.dart';
+import '../session.dart';
+
+/// Middleware that handles secure communication
+Future<dynamic> encryptionMiddleware(
+  dynamic session,
+  dynamic message,
+  MiddlewareAction action,
+  Future<dynamic> Function(dynamic) next,
+) async {
+  final CryptoProvider provider = session.storage['crypto_provider'] ?? CryptoProvider(session);
+
+  // Redirect handshaking messages to the crypto provider
+  if (_isHandshakeMsg(message)) {
+    // Received handshaking messages are captured
+    if (action == MiddlewareAction.receive) {
+      provider(message);
+      return null;
+    }
+
+    // Handshaking messages are not blocked nor passed to other middleware
+    return message;
+  }
+
+  // Wait for secure channel to be established
+  await provider.established;
+
+  if (action == MiddlewareAction.receive) {
+    // We catch all errors here as we can't trust the incoming data
+    try {
+      // Decrypt the message so other middleware can work with it
+      message = await provider.decrypt(message);
+    } catch (e) {
+      session.raise(e);
+      return null;
+    }
+
+    return await next(message);
+  } else {
+    // Let other middleware work with the message, finally encrypt it
+    return await provider.encrypt(await next(message));
+  }
+}
+
+/// Check if message belongs to handshaking protocol
+bool _isHandshakeMsg(String msg) {
+  return msg.startsWith(_CryptoVars.prefix);
+}
+
+/// Class that handles secure communication
+/// between client and server.
+///
+/// Every session has it's own instance of this class.
+/// It normally lives in `session.storage['crypto_provider']`
+class CryptoProvider {
+  /// Algorithm used for key exchange
+  static final X25519 keyExchangeAlgo = Cryptography.instance.x25519();
+
+  /// Algorithm used for encryption
+  static final AesGcm encryptionAlgo = Cryptography.instance.aesGcm();
+
+  /// This future finishes when the handshaking is done and a secure connection is established
+  Future<void> get established => _completer.future;
+
+  /// Completer that allows awaiting secure connection
+  final Completer<void> _completer = Completer<void>();
+
+  /// Future that completes once the own key is available
+  late final Future<void> _ready;
+
+  /// Own public key as string
+  late final String _publicKeyStr;
+
+  /// The session this provider is handling
+  final Session _session;
+
+  /// Our own key pair (private + public)
+  late SimpleKeyPair _ownKeyPair;
+
+  /// The generated shared key
+  SecretKey? _sharedKey;
+
+  CryptoProvider(this._session) {
+    // Install ourselves to the session
+    _session.storage['crypto_provider'] = this;
+
+    // Start handshaking on object creation
+    _ready = _initHandshake();
+  }
+
+  //* General use
```

Comment:
I think you mean [this](https://github.com/Coronon/flyde/commit/97aece826f8857f91d922ff4cff2843887835cbb#diff-1819eceb5d04dd907948b7c95724f7d4eb07001b52ff1642d0f6dbedc8bbf5fd) change, which I obviously fixed. If I marked anything else as resolved it was probably on accident. I don’t like your toxic comment.

Reply:
No I wasn't talking about this, but [this](https://github.com/Coronon/flyde/pull/2#discussion_r671731304)

Reply:
I see, but with only the exports empty lines would really impair visual hierarchy even more. Thats why I closed that one, because it only consists out of `export`‘s

Reply:
This one is fixed